src/main/kotlin/nl/avisi/structurizr/site/generatr/site/DiagramGenerator.kt

```diff
@@ -34,22 +36,18 @@ fun generateDiagrams(workspace: Workspace, exportDir: File) {
         }
 }
 
-fun generateDiagramWithElementLinks(workspace: Workspace, view: View, url: String, exportDir: File): String {
-    val pumlDir = pumlDir(exportDir)
-    val svgDir = svgDir(exportDir)
-
+private val diagramCache = ConcurrentHashMap<String, String>()
+fun generateDiagramWithElementLinks(workspace: Workspace, view: View, url: String): String {
     val diagram = generatePlantUMLDiagramWithElementLinks(workspace, view, url)
 
     val name = "${diagram.key}-${view.key}"
-    val plantUMLFile = File(pumlDir, "$name.puml")
-    if (!plantUMLFile.exists() || plantUMLFile.readText() != diagram.definition) {
-        saveAsSvg(diagram, svgDir, name)
-        saveAsPUML(diagram, plantUMLFile)
```

Comment:
Are you sure that we no longer need the `svg` and `puml` files on disk? I seem to remember that we use those files for the `svg` or `puml` download links in the footer of an embedded svg.

Reply:
As far as I can tell, these files are really only temporary. I've also tested that the "regular" download links still work. The links in the footer link to the PUML and SVG files that are separately generated in the [generateDiagrams function](https://github.com/avisi-cloud/structurizr-site-generatr/blob/main/src/main/kotlin/nl/avisi/structurizr/site/generatr/site/DiagramGenerator.kt#L16).

Reply:
ah yeah, i should have looked at the complete file and not just the diff, thanks for clarifying!